syncer/syncer_test.go

```diff
@@ -87,6 +87,38 @@ func TestSyncer(t *testing.T) {
 	}
 }
 
+func TestParallelSync(t *testing.T) {
```

Comment:
Isn't this just a portion of copy pasta from `TestSyncer` with more blocks being mined to get past the require height? Might as well just delete `TestSyncer` and name this `TestSyncer` instead?

The kind of testing I miss is the not so happy path kind. Where we sync the most recent blocks successfully for example and then the earlier blocks fail and then check that our node isn't completely screwed. 
Maybe then connect to another syncer that has the earlier blocks and see if that finishes the sync successfully.